docs/11-client-side-architecture.md

```diff
@@ -0,0 +1,88 @@
+the [ManagedForm](../public/video-ui/src/components/ManagedForm/ManagedForm.jsx) components. This can present difficulty
+when save requests result in failure or edits are cancelled. In these cases there is no previous state to fall back to
+as edits have been immediately applied. The application also orchestrates saves via Thunks, which are often made up of
+several sequenced fetch requests, if parts of the process fail, it can be unclear what action users should take to
+respond.
+
+Save requests often update the entire atom model, this limits the amount users can work concurrently on atoms.
+Therefore, users rely on Presence indicators to determine whether it is safe to work on an atom. A mechanism exists to
+[report and reject potentially conflicting writes](https://github.com/guardian/media-atom-maker/blob/0d0535c5d6158a4b077c4b23c8df01790be61071/app/model/commands/UpdateAtomCommand.scala#L77-L85),
+but it doesn't work for all endpoints.
+
+## Improvements
+
+Work has been undertaken to improve the codebase in an effort to reduce the number of bugs and improve the overall UX.
+
+### Redux Toolkit
+
+The Redux constructs have been [rewritten using Redux Toolkit](https://github.com/guardian/media-atom-maker/issues/1262)
+in order to make the code more concise and the state and interactions of actions easier to reason about. Reducers and
+Actions have been converted to TypeScript as slices following
+the [RTK migration guide](https://redux.js.org/usage/migrating-to-modern-redux). To a lesser extent components have been
+updated, but this has been limited by the use of React Component Classes and JavaScript.
+
+### TypeScript Coverage
+
+The project's overall TypeScript coverage has increased significantly, roughly doubling during the course of this work.
+This has helped to identify potential bugs, improve the understanding and readability of the code, and generally make
+refactoring safer.
+
+## Future development
+
+When developing the client further, a few key areas should be considered for improvement:
+
+### Function Components
+
+Many React components are written in the now outdated Component Class style. Rewriting these as function components will
+make them more familiar to modern React developers. It also allows the use of hooks which work well with Redux and
+RTK.
+
+### Strict TypeScript
+
+Continuing to improve the TypeScript coverage will make the project easier to work on in the future and provides
+significant benefits to the Developer Experience.
+
+[Setting TypeScript to be `strict`](https://www.typescriptlang.org/docs/handbook/2/basic-types.html#strictness) would be a sensible improvement and help identify further potential bugs.
+
+### Test coverage
+
+The client is sparsely tested, but some examples do exist. When rewriting components, it would be worth adding test
+coverage to ensure consistent behaviour and help document business logic.
```

Comment:
Would it be worth putting an example (such as this one https://github.com/guardian/media-atom-maker/blob/ccfd5c194d2c76e55d0c4b468d91e252716a0878/public/video-ui/src/components/FormFields/TagPicker.test.jsx) here?